### PR TITLE
Full Site Editing: hide customizer if FSE is active

### PR DESCRIPTION
**Pull request by mattwiebe**
#### Changes proposed in this Pull Request

* introduce `isSiteUsingFullSiteEditing` selector
* use it to hide the Customizer sidebar link for sites with Full Site Editing active

#### Testing instructions

* sandbox `public-api.wordpress.com`
* apply D31148-code
* select a site in Calypso with the `full-site-editing` blog sticker and the Modern Business theme
* (you may need to clear your cache)
* you should see the Customizer link hidden in the sidebar:

image

Partial resolution of #34791
Supercedes #35035

### CODE REVIEW

File: client/state/selectors/test/is-site-using-full-site-editing.js [hunk 1]
```diff
@@ -0,0 +1,57 @@
+/** @format */
+
+/**
+ * Internal dependencies
+ */
+import isSiteUsingFullSiteEditing from '../is-site-using-full-site-editing';
+
+describe( 'isSiteUsingFullSiteEditing', () => {
```
Comment (gwwar) [inline client/state/selectors/test/is-site-using-full-site-editing.js]: Noice! ✨ Thanks for adding tests

Comment (mattwiebe) [reply client/state/selectors/test/is-site-using-full-site-editing.js]: Thanks for the nudge - I get lazy sometimes 🤪

File: client/state/selectors/is-site-using-full-site-editing.js [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+/**
+ * Checks if a site is using the new Full Site Editing experience
+ * @param {Object} state  Global state tree
+ * @param {Object} siteId Site ID
+ * @return {Boolean} True if the site is using Full Site Editing, otherwise false
+ */
+export default function isSiteUsingFullSiteEditing( state, siteId ) {
+	const site = getRawSite( state, siteId );
+	const isActive = get( site, 'is_fse_active', false );
+	return isActive && hasStaticFrontPage( state, siteId );
```
Comment (gwwar) [inline client/state/selectors/is-site-using-full-site-editing.js]: Do we also want to gate on static front page? Folks are still able to pick a blog page with this value. I'll add a card for blog page support. (If we pick a page as a blog page, it ignores any post_content in it)

File: client/state/selectors/test/is-site-using-full-site-editing.js (hunk 1 above)
Comment (gwwar) [reply client/state/selectors/test/is-site-using-full-site-editing.js]: I usually only ask for state related pieces, as units are a great fit for catching bugs/testing interfaces.

Component tests/e2e is more of a judgement call, for usefulness + critical path vs overhead.

File: client/state/selectors/is-site-using-full-site-editing.js (hunk 2 above)
Comment (gwwar) [reply client/state/selectors/is-site-using-full-site-editing.js]: Added https://github.com/Automattic/wp-calypso/issues/35107 as a follow up

**Comment by gwwar** [comment]
Getting a test failure from:
> FAIL  client/state/selectors/test/is-site-using-full-site-editing.js

File: client/state/selectors/test/is-site-using-full-site-editing.js [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+	test( 'returns false if site does not exist', () => {
+		const isFSE = isSiteUsingFullSiteEditing( {}, 1 );
```
Comment (gwwar) [inline client/state/selectors/test/is-site-using-full-site-editing.js]: Try passing in `undefined` (not initialized) or mock more of the state structure. The sites reducer should add that prop after the reducer is called.

Comment (gwwar) [reply client/state/selectors/test/is-site-using-full-site-editing.js]: Another option is bulletproofing getRawSites for the case, but I'll leave it up to you!

**Pull request merged**